tv-shows-app/src/components/shared/SidebarNavigation/SideBarNavigation.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { Flex, Text } from "@chakra-ui/react";
+import NextLink from "next/link";
+import { useEffect, useState } from "react";
+import { PiTelevisionSimple } from "react-icons/pi";
+import { FiLogOut } from "react-icons/fi";
+
+export default function SidebarNavigation() {
+  const link = window.location.href;
+  const currentCat = link.split("/").reverse()[0];
+
+  const [category, setCategory] = useState(currentCat);
+
+  useEffect(() => {
```

Comment:
Finally, something that I can really comment on :)

Next.js provides you with the usePathname hook : https://nextjs.org/docs/app/api-reference/functions/use-pathname

This way you don't have to have an effect that will add popstate listeners that will then update the category state. But it is very nice that you implemented it manually. I'd rework it though.